### PR TITLE
disabling 0-dim cuda tensor reduction/normalization

### PR DESCRIPTION
**Pull request by jjsjann123**
1. Disables normalizatoin/reduction for 0-dim cuda inputs.
2. Patches fusion logic on reduction parsing rule

We are doing this because normalization for 0-dim behavior is tricky to justify (?!) and seems implementation specific. So we just leave this for eager.

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/parser.cpp [hunk 1]
```diff
@@ -2052,13 +2072,18 @@ class IrParser {
               // We can only handle output as half, float, and double;
               if (const auto opt_ivalue = toIValue(node->input(3))) {
                 const auto scalar_type = opt_ivalue->toScalarType();
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/parser.cpp:2075]: Maybe worth making a simple helper function since this logic looks repeated frequently.

**Review by csarofeen — APPROVED** [review]
LGTM

File: torch/csrc/jit/codegen/cuda/parser.cpp (hunk 1 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/parser.cpp:2075]: Sounds good. I'll merge those together.

**Comment by jjsjann123** [comment]
Tests passed I'm merging this guy!

**Pull request merged**